attest/verifier/config/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+    NoMeasurementsFound(String),
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    use hex_literal::hex;
+
+    const TEST_DATA: &str = r#"{
+    "v3": {
+       "consensus": {
+           "MRENCLAVE": "207c9705bf640fdb960034595433ee1ff914f9154fbe4bc7fc8a97e912961e5c",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615"]
+       },
+       "fog-ingest": {
+           "MRENCLAVE": "3370f131b41e5a49ed97c4188f7a976461ac6127f8d222a37929ac46b46d560e",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615"]
+       },
+       "fog-ledger": {
+           "MRENCLAVE": "fd4c1c82cca13fa007be15a4c90e2b506c093b21c2e7021a055cbb34aa232f3f",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615"]
+       },
+       "fog-view": {
+           "MRENCLAVE": "dca7521ce4564cc2e54e1637e533ea9d1901c2adcbab0e7a41055e719fb0ff9d",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615"]
+       }
+    },
+    "v4": {
+       "consensus": {
+           "MRENCLAVE": "e35bc15ee92775029a60a715dca05d310ad40993f56ad43bca7e649ccc9021b5",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615", "INTEL-SA-00657"]
+       },
+       "fog-ingest": {
+           "MRENCLAVE": "a8af815564569aae3558d8e4e4be14d1bcec896623166a10494b4eaea3e1c48c",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615", "INTEL-SA-00657"]
+       },
+       "fog-ledger": {
+           "MRENCLAVE": "da209f4b24e8f4471bd6440c4e9f1b3100f1da09e2836d236e285b274901ed3b",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615", "INTEL-SA-00657"]
+       },
+       "fog-view": {
+           "MRENCLAVE": "8c80a2b95a549fa8d928dd0f0771be4f3d774408c0f98bf670b1a2c390706bf3",
+           "mitigated_hardening_advisories": ["INTEL-SA-00334", "INTEL-SA-00615", "INTEL-SA-00657"]
+       }
+    }
+}"#;
+
+    #[test]
+    fn test_loading() {
+        let tms: TrustedMeasurementSet = serde_json::from_str(TEST_DATA).unwrap();
+
+        assert_eq!(tms.table.len(), 2);
+        let v3 = tms.table.get("v3").unwrap();
+        assert_eq!(v3.len(), 4);
+        let v3_consensus = v3.get("consensus").unwrap();
+        match v3_consensus {
+            StatusVerifierConfig::Mrenclave {
+                MRENCLAVE,
+                mitigated_config_advisories,
+                mitigated_hardening_advisories,
+            } => {
+                assert_eq!(
+                    MRENCLAVE,
+                    &hex!("207c9705bf640fdb960034595433ee1ff914f9154fbe4bc7fc8a97e912961e5c")
+                );
+                assert_eq!(mitigated_config_advisories, &Vec::<String>::default());
+                assert_eq!(
+                    mitigated_hardening_advisories,
+                    &["INTEL-SA-00334", "INTEL-SA-00615"]
+                );
+            }
+            _ => panic!("unexpected"),
+        };
```

Comment:
:shrug: not sure if it's much better, the stray `panic!` feels like a code smell to and breaking the asserts down prevents more information on test failures.
This suggestion would require deriving `PartialEq` on `StatusVerifierConfig`
```suggestion
        let expected_consensus = StatusVerifierConfig::Mrenclave {
            MRENCLAVE: hex!("207c9705bf640fdb960034595433ee1ff914f9154fbe4bc7fc8a97e912961e5c"),
            mitigated_config_advisories: vec![],
            mitigated_hardening_advisories: vec![
                "INTEL-SA-00334".to_string(),
                "INTEL-SA-00615".to_string(),
            ],
        };
        assert_eq!(v3_consensus, &expected_consensus);
```

Reply:
sure that's better, thanks